Tests/Realm.Tests/Sync/SessionTests.cs

```diff
@@ -270,31 +237,388 @@ public void Session_ClientReset_DiscardLocal_ManualResetFallback_InitiateClientR
+                {
+                    originalObj.Strings.RemoveAt(2);
+                });
+
+                // ===== clientB =====
+                var configB = await GetIntegrationConfigAsync(partition, null, "realm456.realm", user);
+                configB.Schema = new[] { typeof(SyncObjectWithRequiredStringList) };
+                var tcsAfterClientResetB = new TaskCompletionSource<object>();
+                var afterCbB = GetOnAfterHandler(tcsAfterClientResetB, (before, after) =>
+                {
+                    var list = after.All<SyncObjectWithRequiredStringList>().Single().Strings;
+                    Assert.That(list.Count, Is.EqualTo(3));
+                    Assert.That(list[0], Is.EqualTo("0"));
+                    Assert.That(list[1], Is.EqualTo("1"));
+                    Assert.That(list[2], Is.EqualTo("3"));
+                });
+                configB.ClientResetHandler = new AutomaticRecoveryHandler(AutomaticRecoveryHandler.Fallback.DiscardLocal)
+                {
+                    OnAfterReset = afterCbB
+                };
+
+                using var realmB = await GetRealmAsync(configB);
+                await WaitForDownloadAsync(realmB);
+
+                var originalObjStr = realmB.All<SyncObjectWithRequiredStringList>().Single().Strings;
+                Assert.That(originalObjStr.Count, Is.EqualTo(3));
+                Assert.That(originalObjStr[0], Is.EqualTo("0"));
+                Assert.That(originalObjStr[1], Is.EqualTo("1"));
+                Assert.That(originalObjStr[2], Is.EqualTo("2"));
+
+                var sessionB = GetSession(realmB);
+                sessionB.Stop();
+
+                realmB.Write(() =>
+                {
+                    originalObjStr.Add("3");
+                });
+
+                // ===== clientA =====
+                var result = await ((PartitionSyncConfiguration)realmA.Config).User.Functions.CallAsync<FunctionReturn>("triggerClientResetOnSyncServer");
+                Assert.That(result.status, Is.EqualTo(FunctionReturn.Result.success));
+
+                sessionA.Start();
+                await tcsAfterClientResetA.Task;
+
+                var tscAfterRemoteUpdateA = new TaskCompletionSource<object>();
+                realmA.All<SyncObjectWithRequiredStringList>().First().PropertyChanged += (sender, eventArgs) =>
+                {
+                    if (eventArgs.PropertyName == nameof(SyncObjectWithRequiredStringList.Strings))
+                    {
+                        var list = realmA.All<SyncObjectWithRequiredStringList>().Single().Strings;
+                        Assert.That(list.Count, Is.EqualTo(3));
+                        Assert.That(list[0], Is.EqualTo("0"));
+                        Assert.That(list[1], Is.EqualTo("1"));
+                        Assert.That(list[2], Is.EqualTo("3"));
+                        tscAfterRemoteUpdateA.TrySetResult(null);
+                    }
+                };
```

Comment:
Then, I think we have a problem because all works fine.
You're right, I should have subscribed to `SubscribeForNotifications`, not to `PropertyChanged`.